### DISCUSSION TITLE
Unable to create a grid

### DISCUSSION THREAD

**Discussion by djdabs** [rexrainbow/phaser3-rex-notes#278] [Q&A] [answered]
Been struggling to create a grid.. Tried both the rex plugina and native phaser3 grid option. Neither of them create a grid in the scene. Any ideas on what I could be missing?

```

     export const ROOM_WIDTH = 3126
     export const ROOM_HEIGHT = 1042


           this.add.grid(
                0,
                0,
                ROOM_WIDTH*2,
                ROOM_HEIGHT*2,
                64,
                64,
                '0xffffff',
                '0xffffff',
                '0xffffff',
                '0xffffff',
            )
            
            const grid = this.rexBoard.add.board({
                grid: this.rexBoard.add.quadGrid({
                    x: 0,
                    y: 0,
                    cellWidth: ROOM_WIDTH,
                    cellHeight: ROOM_HEIGHT,
                    type: 1
                }),
                width: ROOM_WIDTH*2,
                height: ROOM_HEIGHT*2,
                // wrap: true,
                // infinity: true
                // dir: 4
            });
            
            this.rexBoard.createTileTexture(grid, 'tile', '0xffffff', '0xffffff', 5)
```

**Comment by rexrainbow** [comment, marked as answer]
`rexBoard.add.board()` return a [board instance](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/board/#add-board-object), `scene.add.grid` return a [Grid Shape game object](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/shape-grid/#create-shape-object). I guess that you need `scene.add.grid`.

**Reply by djdabs** [reply]
thanks for the reply Rex!

I need the grid to snap objects to while dragging.

I tried this.add.grid and the grid never gets added to the scene. It doesn't show up on screen, just black.. Do i need to enable physics or something for grids? 🤔

**Reply by rexrainbow** [reply]
Check phaser3 version to latest one, and check this [demo](http://labs.phaser.io/view.html?src=src/game%20objects/shapes/grid.js).
